data-prepper-core/src/main/java/org/opensearch/dataprepper/acknowledgements/DefaultAcknowledgementSet.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.acknowledgements;
+
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.event.JacksonEvent;
+import org.opensearch.dataprepper.model.event.EventHandle;
+import org.opensearch.dataprepper.event.DefaultEventHandle;
+import org.opensearch.dataprepper.model.acknowledgements.AcknowledgementSet;
+
+import java.util.concurrent.ExecutorService;
+import java.util.function.Consumer;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.concurrent.Future;
+import java.time.Instant;
+
+public class DefaultAcknowledgementSet implements AcknowledgementSet {
+    private final Consumer<Boolean> callback;
+    private final long expiryTime;
+    private final ExecutorService executor;
+    // This lock protects all the non-final members
+    private final ReentrantLock lock;
+    private boolean result;
+    Map<EventHandle, AtomicInteger> pendingAcknowledgments;
+    Future<Void> callbackFuture;
```

Comment:
Cal be `private`.

Can you use `CompletableFuture` instead?

Reply:
Changed to use ScheduledFuture.